glyphs2fontir/src/erase_open_corners.rs

```diff
@@ -0,0 +1,524 @@
+            path.move_to((190.0, 327.0));
+            path.line_to((199.0, 497.0));
+            path.line_to((199.0, 488.0));
+            path.line_to((32.0, 503.0));
+            path.line_to((190.0, 327.0));
+            path.close_path();
+            path
+        }
+        pub(super) fn largeCrossing() -> kurbo::BezPath {
+            let mut path = kurbo::BezPath::new();
+            path.move_to((378.0, 615.0));
+            path.line_to((344.0, 706.0));
+            path.line_to((444.0, 660.0));
+            path.line_to((281.0, 699.0));
+            path.line_to((378.0, 615.0));
+            path.close_path();
+            path
+        }
+    }
+
+    macro_rules! assert_approx {
+        ($left:expr, $right:expr) => {
+            assert_approx!($left, $right, 1e-4)
+        };
+        ($left:expr, $right:expr, $accuracy:literal) => {
+            assert!(
+                ($left - $right).abs() < $accuracy,
+                "{} !~= {}",
+                $left,
+                $right
+            )
+        };
+    }
+
+    #[test]
+    fn test_empty_glyph() {
+        let glyph = python_test_glyphs::space();
+        let reveresed = glyph.reverse_subpaths();
+        for g in [glyph, reveresed] {
+            assert!(erase_open_corners(&g).is_none());
+        }
+    }
+
+    #[test]
+    fn test_corner_glyph() {
+        let glyph = python_test_glyphs::hasCornerGlyph();
+        let after = erase_open_corners(&glyph).unwrap();
+        let new_pt = after.segments().nth(1).unwrap().end();
+
+        assert_approx!(new_pt.x, 114.5417);
+        assert_approx!(new_pt.y, 191.2080);
+    }
+
+    #[test]
+    fn test_curve_curve_glyph() {
+        let glyph = python_test_glyphs::curvyCornerGlyph();
+        let after = erase_open_corners(&glyph);
+        let new_pt = after.unwrap().segments().next().unwrap().start();
+        // teaching to the test: because our segment splitting impl is different
+        // from fonttools', we get slightly different values here; we might get
+        // a few off-by-ones but that's okay?
+        assert_approx!(new_pt.x, 406.4859, 0.2);
+        assert_approx!(new_pt.y, 104.5666, 0.2);
+    }
+
+    #[test]
+    fn test_double_corner_glyph() {
+        let glyph = python_test_glyphs::doubleCornerGlyph();
+        let after = erase_open_corners(&glyph).unwrap();
+        let &[PathEl::MoveTo(one), PathEl::LineTo(two), PathEl::LineTo(tre), PathEl::LineTo(four), PathEl::LineTo(end), PathEl::ClosePath] =
+            after.elements()
+        else {
+            panic!("wrong path elements: {:?}", after.elements())
+        };
+        assert_eq!(one, Point::new(100., 0.,));
+        assert_eq!(two, Point::new(400., 0.,));
+        assert_eq!(tre, Point::new(400., 400.,));
+        assert_eq!(four, Point::new(100., 400.,));
+        // sanity check
+        assert_eq!(one, end);
+        assert!(erase_open_corners(&glyph.reverse_subpaths()).is_none());
```

Comment:
I'd much rather see one assert_eq for the entire sequence, e.g. `assert_eq!(expected, after.elements())`

Reply:
(here and below)